core/common/src/main/java/alluxio/resource/RWLockResource.java

```diff
@@ -59,4 +59,13 @@ public boolean downgrade() {
     mLock = mRwLock.readLock();
     return true;
   }
+
+  /**
+   * This dummy implementation is just to make sure when a subclass of {@link RWLockResource}
+   * calls {@code super.close()}, {@link LockResource#close()} is called.
+   */
+  @Override
+  public void close() {
+    super.close();
+  }
```

Comment:
I'd assume super calls should chain up all the way. If that's the case, I believe we can just make sure overriders of close() call super.close() and don't bother with dummy code.